lib/api.js

```diff
@@ -1,19 +1,36 @@
 const api = require('@asl/service/api');
 const db = require('@asl/schema');
 
+const redis = require('redis');
+const limiter = require('express-limiter');
+
 const { NotFoundError } = require('./errors');
 
 const Workflow = require('./workflow/client');
 
 const errorHandler = require('./error-handler');
 
 module.exports = settings => {
-  const app = api(settings);
 
+  const app = api(settings);
   const models = db(settings.db);
 
   app.db = models;
 
+  const client = redis.createClient(settings.redis);
+
+  if (settings.limiter && settings.limiter.total) {
+    app.use(
+      limiter(app, client)({
+        lookup: ['user.id'],
+        total: settings.limiter.total,
+        expire: 1000 * 60 * 60,
+        whitelist: req => req.path === '/me',
+        onRateLimited: function (req, res, next) {
+          next({ message: 'Rate limit exceeded', status: 429 });
```

Comment:
I would generally prefer to always pass through an `Error` instance if possible. I'd consider implementing a `RateLimitedError` class similar to the [`NotFoundError` used elsewhere](https://github.com/UKHomeOffice/asl-public-api/pull/45/files#diff-9ccb9b0bdd7e6cd41dfc6512731c2719R7) that will set the message and response code automatically.

Then this code just becomes `next(new RateLimitedError());`